File: ios/Khrysalis/src/views/draw/RadialGradient.actual.swift
//Stub file made with Khrysalis 2 (by Lightning Kite)
import Foundation
import CoreGraphics
import UIKit


//--- newRadialGradient(CGFloat, CGFloat, CGFloat, List<Int>, List<CGFloat>, Shader.TileMode)
public func newRadialGradient(
    centerX: CGFloat,
    centerY: CGFloat,
    radius: CGFloat,
    colors: Array<UIColor>,
    stops: Array<CGFloat>,
    tile: Shader.TileMode
) -> ShaderValue {
    return { context in
        context.drawRadialGradient(
            CGGradient(
                colorsSpace: CGColorSpaceCreateDeviceRGB(),
                colors: colors.map { $0.cgColor } as CFArray,
                locations: stops.map { CGFloat($0) }
            )!,
            startCenter: CGPoint(x: CGFloat(centerX), y: CGFloat(centerY)),
            startRadius: 0,
            endCenter: CGPoint(x: CGFloat(centerX), y: CGFloat(centerY)),
            endRadius: radius,
            options: []
        )
    }
}

File: android/src/main/java/com/lightningkite/khrysalis/views/draw/RadialGradient.actual.kt
package com.lightningkite.khrysalis.views.draw

import android.graphics.Shader
import com.lightningkite.khrysalis.views.geometry.GFloat


fun newRadialGradient(
    centerX: GFloat,
    centerY: GFloat,
    radius: GFloat,
    colors: List<Int>,
    stops: List<GFloat>,
    tile: Shader.TileMode
) = android.graphics.RadialGradient(
    centerX,
    centerY,
    radius,
    colors.toIntArray(),
    stops.toFloatArray(),
    tile
)
